src/main/java/co/rsk/federate/btcreleaseclient/cache/PegoutSignedCacheImpl.java

```diff
@@ -0,0 +1,47 @@
+package co.rsk.federate.btcreleaseclient.cache;
+
+import co.rsk.crypto.Keccak256;
+import com.google.common.annotations.VisibleForTesting;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+class PegoutSignedCacheImpl implements PegoutSignedCache {
+
+  private final Map<Keccak256, Instant> cache = new ConcurrentHashMap<>();
+  private final Duration ttl;
+
+  PegoutSignedCacheImpl(Duration ttl) {
+    this.ttl = ttl;
+  }
+
+  @Override
+  public boolean hasAlreadyBeenSigned(Keccak256 pegoutCreationRskTxHash) {
+    Instant currentTimestamp = Instant.now();
+
+    return Optional.ofNullable(pegoutCreationRskTxHash)
+        .map(cache::get)
+        .map(timestamp -> isValidTimestamp(currentTimestamp, timestamp))
+        .orElse(false);
+  }
+
+  @Override
+  public void put(Keccak256 pegoutCreationRskTxHash) {
+    Instant currentTimestamp = Instant.now();
+
+    Optional.ofNullable(pegoutCreationRskTxHash)
+        .ifPresent(rskTxHash -> cache.putIfAbsent(rskTxHash, currentTimestamp));
+  }
+
+  private boolean isValidTimestamp(Instant currentTimestamp, Instant timestamp) {
```

Comment:
Do you need to receive `currentTimestamp` here as a parameter? Looks like you could just get inside the method calling `Instant.now()` and avoid checking is it's null

Reply:
I believe that yes. `isValidTimestamp` will later be called by the clenpup method from the periodic scheduler. Since it will iterate for each record calling this method I believe it would be a better idea to use the same `currentTimestamp` for each record rather than a new one each time the method is called. That was my reasoning behind it, wdyt?

Reply:
i think thats a very good reason to get the `Instant.now()` each time.
If not, we could end up not cleaning a record that should be cleaned, right? because we will be behind real time

Reply:
I have updated for the method to receive only the timestamp cached. Be advised that when called by the scheduler this will compare each record with a new timestamp making it non consistent per round of execution.